### PR TITLE
Conda cookiecutter for new python projects

### PR DESCRIPTION
**Pull request by postelrich**
This PR implements a new cookiecutter feature conda.  It add a single command `conda cookie cut PROJECT_NAME PROJECT_PATH` to setup a new directory with all the things needed for a python project.  This includes the directory structure, conda recipe, setup.py, setup.cfg, travis.yml, git, GitHub repo creation, versioneer, coverage and more. 

This is a rough POC though there really isn't much to it so opening the PR for discussion and feedback.  

I would still like to add in a simple test, initial tag, setting up travisci and clean up the options and output.  Still needs tests.

### CODE REVIEW

File: conda_build/templates/cookie/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,28 @@
+package:
+  name: {{ name }}
+{% raw %}
+  version: {{ environ.get('GIT_DESCRIBE_TAG', '') }}
```
Comment (msarahan) [inline conda_build/templates/cookie/meta.yaml]: Please do not use this style of Jinja2 variables. Setting defaults kind of breaks the iterative parsing.  See https://github.com/conda/conda-build/issues/1646 for more info.

File: conda_build/cookie.py [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+    r = subprocess.Popen(['git', 'config', '--get', 'user.{}'.format(field)], stdout=subprocess.PIPE).communicate()[0]
+    if r:
+        return r.decode('utf-8').strip()
+    else:
+        r = input("Enter {}: ".format(field)) or ""
+        return r.strip()
+
+
+def run_cmd(cmd):
+    """Run shell command and check if successful."""
```
Comment (msarahan) [inline conda_build/cookie.py]: Please use the utils.cmd_in_env function rather than this new wrapper.

File: conda_build/api.py [hunk 3]
```diff
@@ -281,3 +282,13 @@ def update_index(dir_paths, config=None, force=False, check_md5=False, remove=Fa
 
     for path in dir_paths:
         update_index(path, config, force=force, check_md5=check_md5, remove=remove)
+
+
+def cut_project(name, path, conf=None):
+    cookie = Cookie(name, path, conf=conf)
+    create_project_skeleton(cookie)
+    cookie.initial_commit()
+    python_ver = input("Enter python version [2.7]: ") or '2.7'
```
Comment (msarahan) [inline conda_build/api.py]: Please never, ever require interactive input. Make it a CLI argument and pass it as a keyword arg here.

File: conda_build/cli/main_cookie.py [hunk 4]
```diff
@@ -0,0 +1,34 @@
+import argparse
+import sys
+import conda_build.api as api
+from attrdict import AttrDict
+from conda_build.config import Config
+
+
+def parse_args(args):
+    p = argparse.ArgumentParser()
```
Comment (msarahan) [inline conda_build/cli/main_cookie.py]: Conda build's commands all extend conda. This is good and bad. It makes the system nice and uniform, but it is also very slow. Until we do something else, we should stick to the way things are done now.  However, I'm not at all clear on why this whole PR should be a new conda build feature and not just another cookiecutter recipe that we can point to.

**Comment by msarahan** [comment]
@postelrich how do you want to proceed here?  I'd like to discuss this more.  The concept is very valuable, I'm just not sure whether it should belong in conda-build, or as a cookiecutter template repository.

**Comment by postelrich** [comment]
@msarahan I would proceed by implementing your suggestions and expanding on the features.  As for the proper place this should go, my thought process for adding it to conda-build was that it would be one less package a user has to install but it certainly could be it's own `conda/cookie` repo. What are your thoughts?

Aside from adding additional project initialization features, I also wonder what could be done in terms of project management, if there is any utility to wrapping some common tasks.

**Comment by msarahan** [comment]
I think it's feature creep for it to be in conda-build.  I agree that it would be convenient to dodge the extra package install, but it adds much more code to conda-build, code that is effectively duplicated from the cookiecutter project.  I recommend making a repo that cookiecutter can point to, and then conda-build should document and advertise its availability.

**Comment by postelrich** [comment]
@msarahan pulled it out into its own repo. https://github.com/postelrich/conda-cookie

**Comment by msarahan** [comment]
Dang, I wish I had been in communication with you better.  Our summer intern also pulled it out here:  https://github.com/conda/cookiecutter-conda-python/pull/1

Would you like to add to that PR at all?  Would you prefer to replace that PR?  Let me know.  Either way, I'm happy to host it on the conda repo so that you don't have to maintain it if you don't want to.

**Comment by postelrich** [comment]
@msarahan Can we do a conda/cookie repo?

**Comment by msarahan** [comment]
We can, but I'm not sure we should.  My line of thought in naming cookiecutter-conda-python was based on looking at https://github.com/audreyr/cookiecutter#a-pantry-full-of-cookiecutters - they are divided by language.  I thought we would eventually have things like:

* cookiecutter-conda-python
* cookiecutter-conda-r
* cookiecutter-conda-lua
* cookiecutter-conda-perl

Do you have thoughts on that?

**Comment by postelrich** [comment]
Oh I wasn't even making this as a cookiecutter template, didn't even think about that.  This goes farther by setting up git, conda, github, travis, and more.

**Pull request closed without merging**